device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsMethodsSenderLinkHandler.java

```diff
@@ -0,0 +1,124 @@
+    {
+        super(sender, amqpsLinkStateCallback, linkCorrelationId);
+
+        this.senderLinkTag = getTag(deviceClientConfig, linkCorrelationId);
+        this.senderLinkAddress = getAddress(deviceClientConfig);
+
+        //Note that this correlation id value must be equivalent to the correlation id in the method receiver link that it is paired with
+        this.amqpProperties.put(Symbol.getSymbol(CORRELATION_ID_KEY), Symbol.getSymbol(CORRELATION_ID_KEY_PREFIX + this.linkCorrelationId));
+        this.amqpProperties.put(Symbol.getSymbol(VERSION_IDENTIFIER_KEY), deviceClientConfig.getProductInfo().getUserAgentString());
+    }
+
+    static String getTag(DeviceClientConfig deviceClientConfig, String linkCorrelationId)
+    {
+        String moduleId = deviceClientConfig.getModuleId();
+        String deviceId = deviceClientConfig.getDeviceId();
+        if (moduleId != null && !moduleId.isEmpty())
+        {
+            return SENDER_LINK_TAG_PREFIX + deviceId + "/" + moduleId + "-" + linkCorrelationId;
+        }
+        else
+        {
+            return SENDER_LINK_TAG_PREFIX + deviceId + "-" + linkCorrelationId;
+        }
+    }
+
+    private static String getAddress(DeviceClientConfig deviceClientConfig)
+    {
+        String moduleId = deviceClientConfig.getModuleId();
+        String deviceId = deviceClientConfig.getDeviceId();
+        if (moduleId != null && !moduleId.isEmpty())
+        {
+            return String.format(MODULE_SENDER_LINK_ENDPOINT_PATH, deviceId, moduleId);
+        }
+        else
+        {
+            return String.format(DEVICE_SENDER_LINK_ENDPOINT_PATH, deviceId);
+        }
+    }
+
+    @Override
+    public String getLinkInstanceType()
+    {
+        return "methods";
+    }
+
+    @Override
+    protected MessageImpl iotHubMessageToProtonMessage(Message message)
+    {
+        if (message.getMessageType() == MessageType.DEVICE_METHODS)
+        {
+            MessageImpl protonMessage = super.iotHubMessageToProtonMessage(message);
+            IotHubTransportMessage deviceMethodMessage = (IotHubTransportMessage) message;
+
+            Properties properties;
+            if (protonMessage.getProperties() != null)
+            {
+                properties = protonMessage.getProperties();
+            }
+            else
+            {
+                properties = new Properties();
+            }
+
+            if (deviceMethodMessage.getRequestId() != null)
+            {
+                properties.setCorrelationId(UUID.fromString(deviceMethodMessage.getRequestId()));
+            }
+
+            protonMessage.setProperties(properties);
+
+            Map<String, Object> userProperties = new HashMap<>();
+            if (deviceMethodMessage.getStatus() != null)
+            {
+                userProperties.put(APPLICATION_PROPERTY_KEY_IOTHUB_STATUS, Integer.parseInt(deviceMethodMessage.getStatus()));
+            }
+
+            Map<String, Object> applicationPropertiesMap;
```

Comment:
if this isn't used outside of the if statement below, can it be moved into it?

Reply:
It can be moved in there, good catch